WordPress/src/main/java/org/wordpress/android/viewmodel/activitylog/ActivityLogViewModel.kt

```diff
@@ -177,24 +153,157 @@ class ActivityLogViewModel @Inject constructor(
         this.site = site
         this.rewindableOnly = rewindableOnly
 
-        rewindStatusService.start(site)
-        rewindStatusService.rewindProgress.observeForever(rewindProgressObserver)
-        rewindStatusService.rewindAvailable.observeForever(rewindAvailableObserver)
-
-        activityLogStore.getRewindStatusForSite(site)
-
-        reloadEvents(done = true)
+        reloadEvents(true, currentRestoreEvent)
         requestEventsUpdate(false)
 
         showFiltersIfSupported()
     }
 
+    @VisibleForTesting
```

Comment:
I am so glad that you mentioned that and actually for saying that this is an indication that the code should be extracted into a separate class. This is exactly why I did that!

But, actually in this case I kind of had little luck in avoiding exposing the `reloadEvents(...)` since unfortunately I couldn't easily create all the combinations otherwise, in order to do the necessary refactoring, especially before adding the new solution.

Do you think now that a more simpler solution is in place I could possible make that back into a `private` function and simulate the exact some setup through the `public` entry points? I could probably achieve that but I might also miss some of it cases, not sure.

PS: I would personally prefer to leave it as such, hoping that later on it will be extracted out to a separate class.

Reply:
👍 Thanks for the clarification! I understand now, why/when it was added. I agree with you that keeping it as is, is good enough for now ;)!